test/unit/scan/test_folder_traversal.py

```diff
@@ -0,0 +1,195 @@
+        (tmp_path / "dir" / "file2.txt").write_text("content2")
+        (tmp_path / "dir" / "file3.txt").write_text("content3")
+
+        folder = LocalFolder(str(tmp_path / "dir"))
+        local_paths = folder.all_files(reporter=MagicMock())
+        absolute_paths = [path.absolute_path for path in list(local_paths)]
+
+        assert absolute_paths == [
+            fix_windows_path_limit(str(tmp_path / "dir" / "file1.txt")),
+            fix_windows_path_limit(str(tmp_path / "dir" / "file2.txt")),
+            fix_windows_path_limit(str(tmp_path / "dir" / "file3.txt")),
+        ]
+
+    def test_folder_with_subfolders(self, tmp_path):
+
+        # Create a directory structure below with initial scannig point at tmp_path:
+        # tmp_path
+        # ├── dir1
+        # │   ├── file1.txt
+        # │   └── file2.txt
+        # └── dir2
+        #     ├── file3.txt
+        #     └── file4.txt
+
+        d1 = tmp_path / "dir1"
+        d1.mkdir()
+        (d1 / "file1.txt").write_text("content1")
+        (d1 / "file2.txt").write_text("content2")
+
+        d2 = tmp_path / "dir2"
+        d2.mkdir()
+        (d2 / "file3.txt").write_text("content3")
+        (d2 / "file4.txt").write_text("content4")
+
+        folder = LocalFolder(str(tmp_path))
+        local_paths = folder.all_files(reporter=MagicMock())
+        absolute_paths = [path.absolute_path for path in list(local_paths)]
+
+        assert absolute_paths == [
+            fix_windows_path_limit(str(d1 / "file1.txt")),
+            fix_windows_path_limit(str(d1 / "file2.txt")),
+            fix_windows_path_limit(str(d2 / "file3.txt")),
+            fix_windows_path_limit(str(d2 / "file4.txt")),
+        ]
+
+    @pytest.mark.skipif(
+        platform.system() == 'Windows' and platform.python_implementation() == 'PyPy',
+        reason="Symlinks not supported on PyPy/Windows"
+    )
+    def test_folder_with_symlink_to_file(self, tmp_path):
+
+        # Create a directory structure below with initial scannig point at tmp_path:
+        # tmp_path
+        # ├── dir
+        # │   └── file.txt
+        # └── symlink_file.txt -> dir/file.txt
+
+        (tmp_path / "dir").mkdir()
+
+        file = tmp_path / "dir" / "file.txt"
+        file.write_text("content")
+
+        symlink_file = tmp_path / "symlink_file.txt"
+        symlink_file.symlink_to(file)
+
+        folder = LocalFolder(str(tmp_path))
+        local_paths = folder.all_files(reporter=MagicMock())
+
+        absolute_paths = [path.absolute_path for path in list(local_paths)]
+
+        assert absolute_paths == [
+            fix_windows_path_limit(str(file)),
+            fix_windows_path_limit(str(symlink_file))
+        ]
+
+    @pytest.mark.skipif(
+        platform.system() == 'Windows' and platform.python_implementation() == 'PyPy',
+        reason="Symlinks not supported on PyPy/Windows"
+    )
+    @pytest.mark.timeout(5)  # Set a 5-second timeout for this test
```

Comment:
@kkalinowski-reef  Yes, this is to avoid infinite loops. Given how the code work now, it is probably not necessary, but I wrote the tests first, so I wanted to make sure they finish running. I did not experience any issues with other plugins and CI on Windows runs normally. You think I should remove this?

Reply:
@ppolewicz maybe you have some feedback on this?

Reply:
a test with an infinite loop is easy to detect with -vv, but this is handy and since it's a dev/test dependency, I'm ok with it